### ISSUE TITLE
Problems calling anything in HttpClientExtensions from precompiled Azure Functions

### ISSUE THREAD

**Issue by yellowbrickcode** [DuendeArchive/IdentityModel.PCL#65]
I just run into this issue and thought it was probably worth filing an issue for. I have an Azure Function to call an API that I use Identity Server 3 to secure. As part of the function, I use the IdentityModel library to get the token, then I call the API. 

I was using the SetBearerToken extension on my HttpClient object but that was throwing a 'Method Not Found' exception. I also tried using SetToken but had the same issue. 

The strange part is that it works if you write the Azure Function as a CSX, so I'm not sure the exact reason why it's failing for a precompiled function. I have posted on SO & the Azure Functions forum so hopefully someone else might be able to help shed some light there. The SO post has full code repro too.

SO - https://stackoverflow.com/questions/43367113/precompiled-azure-function-throwing-error-on-httpclientextensions-setbearertoken
Azure Functions forum - https://social.msdn.microsoft.com/Forums/azure/en-US/bbb2c42d-670e-4b28-97e1-026af3628545/problem-with-calling-setbearertoken-from-precompiled-function-but-works-in-csx?forum=AzureFunctions#bbb2c42d-670e-4b28-97e1-026af3628545

**Comment by brockallen** [comment]
If you set the header manually does it work?

**Comment by yellowbrickcode** [comment]
Yeah I replaced the call to SetBearerToken with `apiClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", "mytoken");` and that works fine, so there is an easy workaround. I think the issue is probably on the side of Azure Functions, but I figured it's something other people may come across. If I get anywhere with figuring out why it's happening I'll update here.
